File: weather-demo-app-apple/Weather/Domain/DTO/ForecastItemDTO.swift
//
//  ForecastItemDTO.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//


struct ForecastItemDTO: Decodable {
    let dateText: String
    let weather: [WeatherInfoDTO]
    let main: MainInfoDTO

    enum CodingKeys: String, CodingKey {
        case dateText = "dt_txt"
        case weather
        case main
    }
}

File: weather-demo-app-android/app/src/main/java/com/demo/weather/domain/dto/ForecastItemDto.kt
package com.demo.weather.domain.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ForecastItemDto(
    @SerialName("dt_txt")
    val dateText: String,
    val weather: List<WeatherInfoDto>,
    val main: MainInfoDto
)